src/parse.rs

```diff
@@ -79,6 +84,14 @@ impl<'a> Parser<'a> {
             }
             return Ok((ix, Expr::Alt(children)));
         }
+        // can't have numeric backrefs and named backrefs
+        for escape in self.re.split('\\').skip(1) { // re can't start with a backref
+            if escape.starts_with(char::is_numeric) {
+                if self.named_groups.len() > 0 {
```

Comment:
Let's put this condition outside. If there's no named groups, there's no need to do any of the checking.